pkg/limits/ingest_limits.go

```diff
@@ -345,6 +352,11 @@ func (s *IngestLimits) running(ctx context.Context) error {
 				record := iter.Next()
 				sizeBytes += len(record.Value)
 
+				// Update the estimated consumption lag for the partition.
```

Comment:
It is expected that we will overwrite the gauge a lot of times. Since records are consumed in order, the timestamp delta between adjacent records should be minimal.